searchkit/constraints.py

```diff
@@ -109,578 +107,691 @@ def __repr__(self):
-                search_state = self._seek_next(search_state)
-                if search_state.rc == search_state.RC_ERROR:
-                    offset = 0
-                    self.fd_info.reset()
-                    break
-
-                if (search_state.search_range_end -
-                        search_state.search_range_start) < 1:
-                    # we've reached the end of all ranges but the result in
-                    # undetermined.
-                    if search_state.rc != search_state.RC_FOUND_BAD:
-                        self.fd_info.reset()
-                        offset = 0
-                    else:
-                        offset = search_state.cur_ln
-
-                    break
-
-                # log.debug(search_state)
-                if search_state.rc == search_state.RC_FOUND_GOOD:
-                    # log.debug("seek ended at offset=%s", search_state.cur_ln)
-                    offset = search_state.cur_ln
-                    break
-
-                if search_state.rc == search_state.RC_SKIPPING:
-                    if ((search_state.cur_ln >= search_state.search_range_end)
-                            and (len(search_state.invalid_range) ==
-                                 search_state.search_range_end)):
-                        # offset and pos should still be SOF so we
-                        # make this the same
-                        search_state.cur_ln = 0
-                        self.fd_info.reset()
-                        break
-
-                if self.fd_info.iterations >= len(self.fd_info.markers):
-                    log.warning("exiting seek loop since limit reached "
-                                "(eof=%s)", self.fd_info.eof_pos)
-                    offset = 0
-                    self.fd_info.reset()
-                    break
-        else:
-            log.debug("file %s is empty", self.fd_info.fd.name)
+            if not chunk or len(chunk) == 0:
+                # Reached end of file
+                return SearchState(status=FindTokenStatus.REACHED_EOF,
+                                   offset=len(self))
 
-        if not destructive:
-            self.fd_info.fd.reset()
+            chunk_offset = chunk.find(self.LINE_FEED_TOKEN)
+            if chunk_offset != -1:
+                # We've found the token in the chunk.
+                # As the chunk_offset is a relative offset to the chunk
+                # translate it to file offset while returning.
+                return SearchState(status=FindTokenStatus.FOUND,
+                                   offset=(start_offset + current_offset +
+                                           chunk_offset))
+            # We failed to find the token in the chunk.
+            # Progress the current offset forward by
+            # chunk's length.
+            current_offset = current_offset + len(chunk)
+            attempts -= 1
+
+        # Reached max_iterations and found nothing.
```

Comment:
"max_iterations" no longer refers to a variable. Maybe just replace this line with a log like:

log.warning("reached EOF without finding line feed")

Reply:
done